dipy/align/imaffine.py

```diff
@@ -143,6 +143,8 @@ def set_affine(self, affine):
         if self.affine is None:
             self.affine_inv = None
```

Comment:
Sorry to do small suggestions not relevant to this PR - but how about putting this line after `self.affine = affine` so that `self.affine_inv` is always defined (as None or an affine) even if the inverse raises an error. Otherwise it's set to whatever it was before, which could be confusing.  Probably also worth noting that the method sets `self.affine_inv` in the docstring too.